exp/hubble/schema_factory.go

```
@@ -10,7 +10,22 @@ import (
 )
 
 func makeNewAccountState(state *accountState, change *xdr.LedgerEntryChange) (*accountState, error) {
-	// TODO: Handle account removal.
+	accountRemoved, err := isAccountRemoved(change)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not check removed account")
+	}
+	// If the account has been removed, then we return a nil account state.
+	if accountRemoved {
+		return nil, nil
+	}
+
+	// We should never be given a nil pointer for account state, rather one
+	// to an empty account state. If we are given a nil pointer to state
+	// somehow, we replace it with a pointer to empty account state.
+	if state == nil {
+		state = &accountState{}
```

we should fail fast and return an error here -- if we are given a `nil` account state it means we have a bug somewhere which is resulting in this unexpected state, which needs to be fixed. we should not paper over these errors by automatically handling them -- i.e. we should clearly write the code such that a `nil` state is invalid.

Another way of looking at this is that a developer reading this code could reasonably assume that a `nil` value for `state` is allowed and may intentionally pass in `nil` values when they could instead have easily passed in an empty `accountState`. we will then be left with two compatible representations of an empty `accountState` which is strictly worse than having one valid representation that everyone follows.